### DISCUSSION TITLE
Guard on a specific spawned actor instance

### DISCUSSION THREAD

**Discussion by claidler** [statelyai/xstate#3437] [Q&A]
When spawning actors on a machine we pass a name. Im wondering if there is any way to refer to that name in the actor machine when setting up a guard on an action.

For instance (pseudo code):

```
// Parent machine invokes actor machines on some state:


nameInput: spawn(formInputMachine, 'name');
ageInput: spawn(formInputMachine, 'age');


// Child machine event with guard

CHANGE_FORM_VALUE: {
    action: ['changeFormValue'],
    cond: (ctx, evt, meta) => // logic here to check machines spawned name ('age' or 'name') - can we do this?
}

```

Currently I'm passing in the name as an event value, but I'm wondering if this is an option also?

Thanks

**Comment by Andarist** [comment]
Currently, there is no clear way to do this. We could consider adding the `invokeeId` to the `meta`, although I would have to think more to figure out how this could be typed (unless we'd just type it as a `string`).

**Reply by claidler** [reply]
Thanks for getting back to me @Andarist. Is it something you guys are thinking to implement? Or a purposeful design decision not to include it.. or just not something anyone is really asking for?

**Reply by Andarist** [reply]
I was thinking about this in the past - although mainly from the perspective of invokes and not spawns. We just never got to this yet. It's not a purposeful design limitation, just something that has not been requested too often (but I recall similar requests in the past).
